### PR TITLE
Add [PingPong] Service

### PR DESCRIPTION
**Pull request by bluecap-se**
Adds two new badges for [PingPong Status Page](https://pingpong.one) service:

- Uptime (last 30 days)
  Uptime
- Status
  Uptime

### CODE REVIEW

**Comment by calebcartwright** [comment]
hi, thanks for the PR! Could you take a look and address the failing CI checks?

They seem to be linter related, and you should be able to reproduce locally via `npm run lint`

**Comment by bluecap-se** [comment]
Thanks for your feedback @calebcartwright! Please have another look after the fixes.

**Comment by calebcartwright** [comment]
Can you tell us a little more about the API token? I assume that it's absolutely necessary in order to fetch the data?

Can this token have it's permissions/auth. scoped to a read-only mode, or does it inherently provide other permissions as well?

**Comment by bluecap-se** [comment]
Of course. The API token is a public read-only token that's specific for status/uptime badges, making it safe for public sharing. It's required to the fetch the data, otherwise the backend returns 404. It has no other permissions and is separate from the API key with full permissions that's used for the [REST API](https://developers.pingpong.one/).

File: services/pingpong/pingpong-uptime.service.js
Comment (calebcartwright) [inline services/pingpong/pingpong-uptime.service.js:2]: Same general theme for this service class as the status one

File: services/pingpong/pingpong-uptime.tester.js
Comment (calebcartwright) [inline services/pingpong/pingpong-uptime.tester.js:5]: Same general theme for this tester as the one for the status tester. Another benefit of using the standard/consistent Shields approach for setting badge attributes is that we'll control what the message can be, and can accordingly reuse our test validator schemas

**Review by calebcartwright — COMMENTED** [review]
Thanks for the PR and additional info! It seems reasonable to incorporate these badges, though I think we've got an opportunity to simplify the implementation somewhat by reusing some standard Shields helpers.

Have left specific details inline

File: services/pingpong/pingpong-status.service.js
Comment (bluecap-se) [reply services/pingpong/pingpong-status.service.js:15]: Users are presented with the API key on that page, without further action.

File: services/pingpong/pingpong-status.service.js
Comment (bluecap-se) [reply services/pingpong/pingpong-status.service.js:48]: Very good points. In the update, I've referenced [uptimerobot](https://github.com/badges/shields/blob/master/services/uptimerobot/uptimerobot-status.service.js#L25) as it seemed to match this use case; nodeping example doesn't have all status types (`issues`, `maintenance` being the case here) but it pointed me in the right direction - thanks for that! If there's a helper function in Shields that supports all status types, I'd be happy to apply it here.

File: services/pingpong/pingpong-status.service.js
Comment (calebcartwright) [inline services/pingpong/pingpong-status.service.js:8]: I know that the we've changed to a different schema (`status` vs `message` + `color`) which is good, though my prior point about utilizing the schema for validation still holds true. I'd like to see the `status` value here in the schema restricted to the known/supported statuses, or if we're going to keep this wide open to accept any string then we need to fail loudly (throw an `InvalidResponse`) in the default switch case in the render function.

File: services/pingpong/pingpong-status.service.js
Comment (calebcartwright) [inline services/pingpong/pingpong-status.service.js:52]: This pairs with my comment above on the schema

Any occurrence that hits this branch would mean that something's changed upstream that this service class in Shields isn't prepared to handle. That's an error scenario that we want to acknowledge loudly, whereas returning a non-error badge with an `unknown` message could give the user the incorrect impression that PingPong doesn't know the status of their site

File: services/pingpong/pingpong-uptime.tester.js
Comment (calebcartwright) [inline services/pingpong/pingpong-uptime.tester.js:9]: I feel like we should be able to replace this with our `isPercentage` helper from `test-validators` and avoid having another  regex to maintain. Could you give that a try, or explain why we need a separate one here?

File: services/pingpong/pingpong-uptime.tester.js
Comment (calebcartwright) [inline services/pingpong/pingpong-uptime.tester.js:27]: Thanks again for the thoroughness of the tests but please remove this one as it's unnecessary and duplicative

**Review by calebcartwright — COMMENTED** [review]
Thank you for the changes and apologies for the extended delay on the second review; this one fell off the radar!

This is in much better state now, though with a few comparatively minor things that'll need to be adjusted before merging

File: services/pingpong/pingpong-status.service.js
Comment (bluecap-se) [reply services/pingpong/pingpong-status.service.js:8]: Sounds good. In order to keep the checks DRY, the render function now throws an error.

**Comment by calebcartwright** [comment]
Updates look good thanks! Could you rebase on master? We have the up-to-date branch gate enabled so will need to get this in sync with master before it can be merged, but should be good to go afterwards

**Comment by PyvesB** [comment]
Great to see this work completed, thanks @bluecap-se  and @calebcartwright for your work! :)

**Comment by calebcartwright** [comment]
Oops, looks like another PR got merged inbetween and this is out of date again. Will require another rebase on your end sorry. IIRC you should also be able to check the box to allow maintainers to update the PR which would allow us to sync the branches for you and make it a bit easier to merge

**Comment by bluecap-se** [comment]
@calebcartwright The branch should be updated again. The option to allow maintainers doesn't seem to exist on this PR.

**Review by PyvesB — APPROVED** [review]
Strange that the option doesn't exist on this PR. Let's get this merged before it's too late again! :)

**Comment by PyvesB** [comment]
These badges are now live! For example: https://img.shields.io/pingpong/status/sp_2e80bc00b6054faeb2b87e2464be337e 



🍾

**Comment by bluecap-se** [comment]
@calebcartwright @PyvesB Great to see this live, thanks for all your feedback and assistance! 🚀 

Our Twitter handle is [@pingpong_status](https://twitter.com/pingpong_status)

As the creator of PingPong status pages I'd like to present our offering to all open source projects: [free Pro plan](https://pingpong.one/pricing/). If this sounds interesting for Shields, start by [creating a Pro account](https://my.pingpong.one/account/signup/?plan=pro) and then [ping us](mailto:support@pingpong.one?subject=Open-source%20setup) to activate the open source offer.

**Comment by bluecap-se** [comment]
Sorry to see https://status.shields.io still being down, let us at [PingPong](https://pingpong.one) know if we can assist in any way.

**Pull request merged**
